pkg/schemaconv/testdata/new-schema.yaml

```diff
@@ -6621,6 +6625,35 @@ types:
     - name: vsphereVolume
       type:
         namedType: io.k8s.api.core.v1.VsphereVirtualDiskVolumeSource
+    union:
+      fields:
```

Comment:
Sorry to leave this comment here and not in an earlier change, but...
would this make more sense as "choices", where it's a list of structs which set "field" and "discriminatedBy" or something? e.g.

```
union:
  discriminator: type
  options:
  - fieldName: fc
    discriminatedBy: FC
  - fieldName: azureDisk
    discriminatedBy: AzureDisk
...
```

I think it is more understandable that way, what do you think?

Reply:
You could even indicate a "none" option that way by permitting a single entry to have no fieldName and just a discriminatedBy.

(no strong opinion about the exact names intended)

Reply:
Yeah, Jenny made a similar comment on the other PR, I agree. That's also much more kubernetes-style.